### ISSUE TITLE
Multiple page selections

### ISSUE THREAD

**Issue by jflesch** [openpaperwork/paperwork#132]
Currently, pages can be dragged-and-dropped and later, it will be possible to cut/copy/paste them. However, it's only possible to select one page at a time.

Being able to select multiple pages in one shot would be handy.

**Comment by mjourdan** [comment]
Please find [wireframes here](https://github.com/mjourdan/gnome-design/blob/master/paperwork/wires/selection-mode.png).

**Comment by jflesch** [comment]
Hmm, while I easily see how rotations could be done on many pages at once,
* is there any real use case ?
* How would that go for cropping ?

**Comment by mjourdan** [comment]
Use case: scan a bunch of papers at work, sort them at home. Some of them may be horizontal.

Cropping would cover the same use case, but it requires a more elaborated wireframe.

**Comment by jflesch** [comment]
Hm, I see.

Regarding cropping, my question was more: how do you apply cropping to many pages in one shot, knowing they may not have all the same size or orientation ?

**Comment by mjourdan** [comment]
> Regarding cropping, my question was more: how do you apply cropping to many pages in one shot, knowing they may not have all the same size or orientation ?

Cropping multiple pages is allowed as long as the selected pages are of the same dimensions and orientation. If they don't, a message is shown telling this is not allowed. I updated [the wireframes](https://github.com/mjourdan/gnome-design/blob/master/paperwork/wires/selection-mode.png).

**Comment by jflesch** [comment]
+1 for the "notifications help repair mistakes". I guess it would be a nice way to solve #19 .

Regarding the cropping, wouldn't it make more sense to disable the button, and display a tooltip when the user move the mouse over it, explaining why it's disabled ?
I'm not found of letting the user do something only to tell them that they can't actually do it.

(I also note that I have to improve DnD support)

**Comment by jflesch** [comment]
Questions:
- At which point would we switch into multiple selection mode ?
- Would it be simpler to *not* have the options "crop"/"rotate" in the menu ? Instead we could just have the "goto multiple selection" button that also make the edit tool bar appear ? (which would kind of turn the button "multiple selection" into a "edit page(s)" button I guess)

**Comment by mjourdan** [comment]
Another idea, starting from the goal where the user would like to [export several pages](https://github.com/mjourdan/gnome-design/blob/master/paperwork/experiments/export-pane-selection.jpeg).

**Comment by jflesch** [comment]
Looks good to me too.
Just regarding the "share" part, I'm not sure if it can be done easily (note sure how to start the email writer with an attachment ; and the other button will require extra settings). So I guess this part will come much later.

**Comment by jflesch** [comment]
I'm just not sure how that would fit with your ideas regarding page editing ?  oO

**Comment by mjourdan** [comment]
Late answer, sorry…

So, it could be possible to put several documents and pages to a queue, in order to export them all at once, [like this](https://github.com/mjourdan/gnome-design/blob/master/paperwork/experiments-2.0/export.png). Export means "print, save as, share by mail/nextcloud/whatever".

To edit pages, two versions here. Putting [controls in the header bar](https://github.com/mjourdan/gnome-design/blob/master/paperwork/experiments-2.0/edit-page-bar.png) does not allow for mass editing. An alternative would be to put [controls in a vertical pane](https://github.com/mjourdan/gnome-design/blob/master/paperwork/experiments-2.0/edit-page-pane.png), where the user could first work on a page, and apply the changes to all pages of the document by ticking a checkbox.

**Comment by mjourdan** [comment]
The queue, instead of being accessed through a popover [as seen previously](https://github.com/mjourdan/gnome-design/blob/master/paperwork/experiments-2.0/export.png), could be accessed from the overview pane, as shown in the third variant of the [mockups referenced in #548](https://github.com/mjourdan/gnome-design/blob/master/paperwork/experiments-2.0/filters-overview-pane.png). Maybe this would be more logical?

**Comment by jflesch** [comment]
> in order to export them all at once, [like this](https://github.com/mjourdan/gnome-design/blob/master/paperwork/experiments-2.0/export.png)

That could work. While maybe not intuitive at first, after one or two tries, I guess people will get it.
Just one minor detail: I suggest we display the action buttons beside the document queue instead of below them. I think it would make it clearer that the actions are going to be applied on the queue.

> Putting [controls in the header bar](https://github.com/mjourdan/gnome-design/blob/master/paperwork/experiments-2.0/edit-page-bar.png) does not allow for mass editing.
> An alternative would be to put [controls in a vertical pane](https://github.com/mjourdan/gnome-design/blob/master/paperwork/experiments-2.0/edit-page-pane.png),

IMHO, the alternative looks better. When clicking on "edit", people will either expect a large visible UI change (like the right side of the window changing), or a small change close to their mouse. I don't think people will expect a small change in the header bar.

**Comment by jflesch** [comment]
Regarding the queue, I think the popover version would be good enough (yes, I really don't like sliding panels pushing stuff ;).

**Comment by mjourdan** [comment]
> That could work. While maybe not intuitive at first, after one or two tries, I guess people will get it.

Yeah, I'm trying to use well-known pattern here, so user don't have to learn again how to do the action. The thing is, I'm not sure there is really such pattern. 

 > Just one minor detail: I suggest we display the action buttons beside the document queue instead of below them. I think it would make it clearer that the actions are going to be applied on the queue.

Well, I doubt two columns would help to make things more obvious but I see your point.

> IMHO, the alternative looks better. When clicking on "edit", people will either expect a large visible UI change (like the right side of the window changing), or a small change close to their mouse. I don't think people will expect a small change in the header bar.

Good point.

> Regarding the queue, I think the popover version would be good enough (yes, I really don't like sliding panels pushing stuff ;).

So I'll keep working on both the overview and the multiple page selections because I don't want the popover to be implemented for wrong reasons :D 

FYI, the selection mode (with checkboxes, as discussed earlier in this ticket) does exist in the [GNOME hig](https://developer.gnome.org/hig/stable/) but is not even implemented in Nautilus. There is an [interesting discussion](https://gitlab.gnome.org/GNOME/nautilus/issues/322#use-cases) I'll keep watching as it may provide valuable insights.

**Comment by jflesch** [comment]
> FYI, the selection mode (with checkboxes, as discussed earlier in this ticket) does exist in the GNOME hig but is not even implemented in Nautilus

https://developer.gnome.org/hig/stable/selection-mode.html.en ?
Just so you know, I've seen it in Gnome documents and Gnome music.
